Content.Client/Sprite/SpriteFadeSystem.cs

```diff
@@ -0,0 +1,102 @@
+using Content.Client.Gameplay;
+using Content.Shared.Sprite;
+using Robust.Client.GameObjects;
+using Robust.Client.Player;
+using Robust.Client.State;
+
+namespace Content.Client.Sprite;
+
+public sealed class SpriteFadeSystem : EntitySystem
+{
+    /*
+     * If the player entity is obstructed under the specified components then it will drop the alpha for that entity
+     * so the player is still visible.
+     */
+
+    [Dependency] private readonly IPlayerManager _playerManager = default!;
+    [Dependency] private readonly IStateManager _stateManager = default!;
+
+    private readonly HashSet<FadingSpriteComponent> _comps = new();
+
+    private const float TargetAlpha = 0.4f;
+    private const float ChangeRate = 1f;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<FadingSpriteComponent, ComponentShutdown>(OnFadingShutdown);
+    }
+
+    private void OnFadingShutdown(EntityUid uid, FadingSpriteComponent component, ComponentShutdown args)
+    {
+        if (MetaData(uid).EntityLifeStage >= EntityLifeStage.Terminating || !TryComp<SpriteComponent>(uid, out var sprite))
+            return;
+
+        sprite.Color = sprite.Color.WithAlpha(component.OriginalAlpha);
+    }
+
+    public override void FrameUpdate(float frameTime)
+    {
+        base.FrameUpdate(frameTime);
+
+        var player = _playerManager.LocalPlayer?.ControlledEntity;
+
+        if (!TryComp<TransformComponent>(player, out var playerXform))
+            return;
+
+        if (_stateManager.CurrentState is not GameplayState state)
+            return;
+
+        var spriteQuery = GetEntityQuery<SpriteComponent>();
+        var fadeQuery = GetEntityQuery<SpriteFadeComponent>();
+        var change = ChangeRate * frameTime;
+        var mapPos = playerXform.MapPosition;
+
+        // Also want to handle large entities even if they may not be clickable.
+        foreach (var ent in state.GetEntitiesUnderPosition(mapPos, 6f))
```

Comment:
Contrary to the comment, this only returns clickable entities (though maybe without an interaction outline).  If you want to include non-clickable entities, then clickable & click-maps should be separated out.